feat: implement basic FeeCalculator with single day calculation

yahgw-ai opened the pull request:
## Summary
- Implement FeeCalculator class following issue #194 requirements
- Add comprehensive unit tests following TDD approach
- Calculate fees for first distributor and first date only

## Test plan
- [x] Unit tests for FeeCalculator constructor and basic structure
- [x] Unit tests for calculateFees method behavior
- [x] Tests verify single day calculation with balance_change_wei = "0"
- [x] Tests verify no distribution event processing
- [x] All tests pass
- [x] Pre-commit hooks pass (TypeScript, ESLint, Prettier)

## Implementation Details
This PR implements the minimal FeeCalculator functionality as specified:
- Reads distributor list from FileManager
- Processes only the first distributor
- Reads balance data for that distributor
- Processes only the first date from balance data
- Creates fee report with single day entry where:
  - balance_change_wei is always "0" for first day
  - distributions_wei is "0" (no event processing)
  - total_wei is "0" (0 + 0)
- Writes report using FileManager

The implementation follows TDD principles with RED-GREEN-REFACTOR phases clearly committed.

Closes #194

yahgw-ai committed:
test: add comprehensive failing tests for FeeCalculator behavior

Add complete test suite for FeeCalculator functionality:
- Test reading distributor data from FileManager
- Test processing only first distributor
- Test processing only first date from balance data
- Test writing fee report with correct structure
- Test balance_change_wei is 0 for first day
- Test distributions are 0 (no event processing)
- Test distributorAddress parameter is ignored

Also add minimal FeeCalculator stub to satisfy TypeScript compilation.
All tests are failing as expected in TDD RED phase.

yahgw-ai committed:
feat: implement minimal FeeCalculator to pass all tests

Implement FeeCalculator class following TDD GREEN phase:
- Read distributor list from FileManager
- Process only the first distributor
- Read balance data for that distributor
- Process only the first date from balance data
- Create fee report with single day entry
- Set balance_change_wei to "0" (always for first day)
- Set distributions_wei to "0" and distributions_count to 0
- Calculate total_wei as "0" (0 + 0)
- Write report using FileManager

All tests now pass. Implementation is minimal as per TDD principles.

yahgw-ai committed:
refactor: improve FeeCalculator code clarity and structure

Refactor FeeCalculator implementation following TDD REFACTOR phase:
- Extract constants for magic values
- Use CHAIN_IDS constant from types
- Simplify early returns for better readability
- Add descriptive constant names for fee report values

All tests continue to pass. Code is now more maintainable.

yahgwai commented on __tests__/units/fee-calculator.test.ts:
```diff
@@ -0,0 +1,385 @@
+  FeeReport,
+} from "../../src/types";
+
+jest.mock("../../src/file-manager");
```
Lets not mock the filemanager - lets actually write data to it during the test. We should create and clean up a test data folder at the start/end of the script

yahgw-ai committed:
test: replace unit tests with integration tests for FeeCalculator

Following PR review feedback, replace mocked FileManager tests with
integration tests that use real file operations. This provides better
test coverage and validates actual file I/O behavior.

Tests now:
- Create and clean up temporary test directories
- Write actual test data to FileManager
- Read back and verify fee reports
- Test all edge cases with real file operations
